src/commands/transaction/reconstruct_transaction/mod.rs

```diff
@@ -197,9 +211,25 @@ fn action_transformation(
             )))
         }
         Action::DeployContract(deploy_contract_action) => {
+            // Unfortunately, RPC doesn't return the code for the deployed contract. Only the hash.
+            // So we need to fetch it from archive node.
+
+            let code = crate::commands::contract::download_wasm::get_code(
+                &receiver_id,
+                network_config,
+                block_reference
+            ).map_err(|e| {
+                color_eyre::Report::msg(format!("Couldn't fetch the code. Please use the archival node to fetch the code. Error: {}", e))
+            })?;
+
+            let code_hash = near_primitives::hash::CryptoHash::hash_bytes(&code);
+            if code_hash.0 != deploy_contract_action.code.as_slice() {
+                return Err(color_eyre::Report::msg("The code hash of the contract deploy action does not match the code that we retrieved from the archive node.".to_string()));
+            }
+
             std::fs::write(
                 "reconstruct-transaction-deploy-code.wasm",
-                deploy_contract_action.code
+                code
             )
             .wrap_err("Failed to write the deploy command code to file: 'reconstruct-transaction-deploy-code.wasm' in the current folder")?;
             Ok(Some(add_action::CliActionSubcommand::DeployContract(
```

Comment:
nit: suggest adding this msg here, so that this detail is explicit for anyone seeing this for the first time: 
```rust
            tracing::info!(
                parent: &tracing::Span::none(),
                "The file `{}` with contract code of `{}` was downloaded successfully",
                "reconstruct-transaction-deploy-code.wasm",
                receiver_id,
            );
```